src/ILCompiler.Compiler/src/Compiler/UsageBasedMetadataManager.cs

```diff
@@ -181,6 +185,35 @@ protected override void GetMetadataDependenciesDueToReflectability(ref Dependenc
                     }
                 }
             }
+
+            // If a type is marked [Serializable], make sure a couple things are also included.
+            if (type.IsSerializable && !type.IsGenericDefinition)
+            {
+                foreach (MethodDesc method in type.GetAllMethods())
+                {
+                    MethodSignature signature = method.Signature;
+
+                    if (method.IsConstructor
+                        && signature.Length == 2
+                        && signature[0] == _serializationInfoType
+                        /* && signature[1] is StreamingContext */)
```

Comment:
Oversight or intentional shortcut?

Reply:
Shortcut. I don't think it matters much.